Documentation changes for EC2 spot instances (#432)
giantswarm/docs: src/content/reference/gsctl/create-nodepool.md

    --nodes-max 20
```

> To configure spot instances in your node pool you can set a base capacity of on-demand instances and then a percentage of spot instances used in the node pool:

Reviewer: ```suggestion
To request spot instances in your node pool, you have to set the percentage of worker nodes to be using spot instances via the `--aws-spot-percentage` flag. Optionally you can also define a number of workers that will be guaranteed to use on-demand instances, using the flag `--aws-on-demand-base-capacity`. Here is an example with 3 on-demand instances as a base capacity and 50 percent spot instances above that:
```

Longer, but hopefully easier to understand.

```nohighlight
$ gsctl create nodepool "Cluster name" \
    --name "General purpose m4.2xlarge 3AZ autoscaling" \
    --aws-instance-type m4.2xlarge \
    --num-availability-zones 3 \
    --nodes-min 3 \
    --nodes-max 20 \
    --aws-on-demand-base-capacity 3 \

>     --aws-spot-percentage 50

Reviewer: ```suggestion
    --name "General purpose m4.2xlarge 3AZ spot" \
    --aws-instance-type m4.2xlarge \
    --num-availability-zones 3 \
    --aws-on-demand-base-capacity 3 \
    --aws-spot-percentage 50
```

Simplified the example

```

### Options

- `-n, --name`: Name or purpose description of the node pool. Defaults to "Unnamed node pool".
- `--num-availability-zones`: Number of availability zones to use. Default is `1`. Use this option if you don't care about the specific zones to use.
- `--availability-zones`: Comma-separated list of availability zones to use, instead of setting a number. Use this option if you care about the exact zones to use.
- `--aws-instance-type`: EC2 instance type to use for workers, e. g. `m5.2xlarge`.
- `--aws-on-demand-base-capacity`: Number of on-demand instances that this node pool needs to have until spot instances are used. Defaults to `0`.

> - `--aws-spot-percentage`: Percentage of spot instances used once the on-demand base capacity is fullfilled. A number of 40 would mean that 60% will be on-demand and 40% will be spot instances. Defaults to `0`.

Reviewer: ```suggestion
- `--aws-spot-percentage`: Percentage of spot instances used once the on-demand base capacity is fulfilled. A number of 40 would mean that 60% will be on-demand and 40% will be spot instances. Defaults to `0`.
```
